core/src/main/kotlin/org/evomaster/core/search/algorithms/Neighborhood.kt

```diff
@@ -0,0 +1,92 @@
+package org.evomaster.core.search.algorithms
+
+import org.evomaster.core.search.Individual
+import org.evomaster.core.search.algorithms.wts.WtsEvalIndividual
+import kotlin.math.sqrt
+
+/**
+ * Neighborhood utilities inspired by EvoSuite's Neighbourhood for Cellular GA.
+ * Returns original references (no copies) to be compatible with identity checks in tests.
+ */
+class Neighborhood<T : Individual>(private val populationSize: Int) : NeighborModels<WtsEvalIndividual<T>> {
```

Comment:
have an `init{}` block in which you validate `populationSize`. ie, throw exception if negative (and 0 as well?)